examples/reference/custom_components/ReactiveESM.md

```diff
@@ -0,0 +1,336 @@
+
+```python
+import panel as pn
+import param
+import datetime
+
+pn.extension()
+
+class ButtonEventExample(pn.ReactiveESM):
+
+    value = param.String()
+
+    _esm = """
+    export function render({ data }) {
+        const btn = document.createElement('button')
+        btn.innerHTML = `Click Me`;
+        btn.onclick = (event) => {
+            const currentDate = new Date();
+            const custom_event = new CustomEvent("click", { detail: currentDate.getTime() });
+            data.send_event('click', custom_event)
+            }
+        return btn
+    }
+    """
+
+    def on_click(self, event):
+        unix_timestamp = event.data["detail"]/1000
+        python_datetime = datetime.datetime.fromtimestamp(unix_timestamp)
+        self.value = str(python_datetime)
+
+button = ButtonEventExample()
+pn.Column(
+    button, button.param.value,
+).servable()
+```
+
+## Dependency Imports
+
+JavaScript dependencies can be directly imported via URLs, such as those from [`esm.sh`](https://esm.sh/).
+
+```python
+import panel as pn
+
+pn.extension()
+
+class ConfettiButton(pn.ReactiveESM):
+
+    _esm = """
+    import confetti from "https://esm.sh/canvas-confetti@1.6.0";
+
+    export function render() {
+        let btn = document.createElement("button");
+        btn.innerHTML = `Click Me`;
+        btn.addEventListener("click", () => {
+            confetti()
+        });
+        return btn
+    }
+    """
+
+ConfettiButton().servable()
+```
+
+Use the `_import_map` attribute for more concise module references.
+
+```python
+import panel as pn
+
+pn.extension()
+
+class ConfettiButton(pn.ReactiveESM):
+    _import_map = {
+        "canvas-confetti": "https://esm.sh/canvas-confetti@1.6.0",
+    }
+
+    _esm = """
+    import confetti from "canvas-confetti";
+
+    export function render() {
+        let btn = document.createElement("button");
+        btn.innerHTML = `Click Me`;
+        btn.addEventListener("click", () => {
+            confetti()
+        });
+        return btn
+    }
+    """
+
+ConfettiButton().servable()
+```
+
+## External Files
+
+You can load JavaScript and CSS from files by providing the paths to these files.
+
+Create the file **counter_button.py**.
+
+```python
+from pathlib import Path
+
+import param
+import panel as pn
+
+pn.extension()
+
+class CounterButton(pn.ReactiveESM):
+
+    value = param.Integer()
+
+    _esm = Path("counter_button.js")
+    _stylesheets = [Path("counter_button.css")]
+
+CounterButton().servable()
+```
+
+Now create the file **counter_button.js**.
+
+```javascript
+export function render({ data }) {
+    let btn = document.createElement("button");
+    btn.innerHTML = `count is ${data.value}`;
+    btn.addEventListener("click", () => {
+        data.value += 1;
+    });
+    data.watch(() => {
+        btn.innerHTML = `count is ${data.value}`;
+    }, 'value');
+    return btn;
+}
+```
+
+Now create the file **counter_button.css**.
+
+```css
+button {
```

Comment:
When I edit the css file I see the button reloading. But the styling is not changed.

Reply:
Ah, indeed, seems like it's actually loading it from a server endpoint instead of inlining it. Will fix.

Reply:
Fixed.